addons/sale/models/payment_transaction.py

```diff
@@ -125,8 +125,11 @@ def _reconcile_after_done(self):
             self._invoice_sale_orders()
         super()._reconcile_after_done()
         if auto_invoice:
-            # Must be called after the super() call to make sure the invoice are correctly posted.
-            self._send_invoice()
+            if str2bool(self.env['ir.config_parameter'].sudo().get_param('sale.async_emails')):
+                self.env.ref('sale.send_invoice_cron')._trigger()
```

Comment:
Two situations could be problematic:
1) the cron is deleted
2) the cron is inactive (AFAICS, it creates the trigger but won't run the cron in that situation)

I believe that in those two situations, we should still call `_send_invoice`.